### PR TITLE
Add thank you to community authored changes.

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
Update changelog generation to add a thanks with link to the author of a community authored change. Lit team members are filtered out.

For example when I removed `augustine` from the filter, `npm run version` locally generated the following changelog entry:

### Sample entry with thank you

[#3905](https://github.com/lit/lit/pull/3905) [f8d72859](https://github.com/lit/lit/commit/f8d7285976390bbb35c540eb7b516f3748064d19) Thanks [@augustjk](https://github.com/augustjk)! - Handle non-array iterables including map directive in child parts

### Multiple authors

Tested changesets that include multiple authors with a lit core member.

```
author:testone
author:testtwo
author:AndrewJakubowicz
```

This will render:

```
- [#3901](https://github.com/lit/lit/pull/3901) [`82e9f370`](https://github.com/lit/lit/commit/82e9f3708973709e916ccbb6d8a450110dea7755) Thanks [@testone](https://github.com/testone), [@testtwo](https://github.com/testtwo)! - <Description>
```

### Multiple authors with only lit core

```
author:AndrewJakubowicz
```

Does not render any Thanks.

### CODE REVIEW

**Review by augustjk — APPROVED** [review]

File: .changeset/changelog-lit.cjs
Comment (usergenic) [inline .changeset/changelog-lit.cjs:130]:
```ts
return new RegExp(`\\b(${[
        'AndrewJakubowicz',
        'augustjk',
        'bicknellr',
        'dfreedm',
        'e111077',
        'justinfagnani',
        'kevinpschaaf',
        'rictic',
        'sorvell',
        'usergenic',
      ].join('|')})\\b`, 'i').test(s);
```

I went down a rabbit-hole and overthought the problem of usernames that might be supersets of our usernames :-P

LGTM tho.

**Review by usergenic — APPROVED** [review]

File: .changeset/changelog-lit.cjs
Comment (augustjk) [reply .changeset/changelog-lit.cjs:130]: hahaha that's a great point. impersonators don't deserve thanks!

Comment (AndrewJakubowicz) [reply .changeset/changelog-lit.cjs:130]: Applied and tested this fantastic solution! Thank you! We definitely want to thank `rictickler` or `chadfreeman` if they contribute!

File: .changeset/changelog-lit.cjs
Comment (justinfagnani) [inline .changeset/changelog-lit.cjs:102]: This removes any mention of the user, yeah? I don't think that's what we want. I think we just want to say "Thanks" only to the external contributors.

**Review by justinfagnani — CHANGES_REQUESTED** [review]

File: .changeset/changelog-lit.cjs
Comment (augustjk) [reply .changeset/changelog-lit.cjs:102]: line 139 below has the `users` check that conditionally adds the whole "Thanks" phrase with the users' github handle linked.
```ts
    const prefix = [
      links.pull === null ? '' : ` ${links.pull}`,
      links.commit === null ? '' : ` ${links.commit}`,
      users === null ? '' : ` Thanks ${users}!`,
    ].join('');
```
If user is a lit team member, `users` would be null.

Or are you saying you want to just say "Thanks" and not have the user's github handle linked?

Comment (AndrewJakubowicz) [reply .changeset/changelog-lit.cjs:102]: Ah this is a case that our current changesets are not covering. In this code path that isn't exercised the result of filtering out team members would result in an empty string. An empty string would fail to match with `null` and would print an empty `Thanks`.

I will make the null handling also check for an empty string & locally test this code path.

Comment (AndrewJakubowicz) [reply .changeset/changelog-lit.cjs:102]: Fixed by also checking for an empty string & manually tested the cases with multiple `author:<author>` written in a changeset file.

Comment (AndrewJakubowicz) [reply .changeset/changelog-lit.cjs:102]: Understood! Although I misunderstood you, you caught a real bug that could have given unattributed thanks! 😄 

Current behavior is that all changes ignore users & have no attribution or thanks.

This PR: Adds attribution & thanks to external contributors, but core members continue to receive no thanks or attribution.

Your proposal is to always add attribution, but only add the additional prefixed "Thanks <contributor list>!" if it's an external contributor?
Would you like me to change to always include attribution?

Comment (augustjk) [reply .changeset/changelog-lit.cjs:102]: Ooh I totally misunderstood that. I'm less sure if we need the attribution for team members since clicking on the PR link right before it would show who worked on it. But no strong feelings either way.

**Comment by AndrewJakubowicz** [comment]
This is good to go pending a review from Justin. And we have some community contributions queued up in the release pipeline!

**Review by rictic — APPROVED** [review]

**Pull request merged**
